applications/serial_lte_modem/src/slm_at_host.c

```diff
@@ -192,6 +236,77 @@ static int handle_at_sleep(const char *at_cmd, enum shutdown_modes *mode)
 	return ret;
 }
 
+static int handle_at_slmuart(const char *at_cmd, u32_t *baudrate)
+{
+	int ret = -EINVAL;
+	enum at_cmd_type type;
+
+	ret = at_parser_params_from_str(at_cmd, NULL, &at_param_list);
+	if (ret < 0) {
+		LOG_ERR("Failed to parse AT command %d", ret);
+		return -EINVAL;
+	}
+
+	type = at_parser_cmd_type_get(at_cmd);
+	if (type == AT_CMD_TYPE_SET_COMMAND) {
+		if (at_params_valid_count_get(&at_param_list) > 1) {
+			ret = at_params_int_get(&at_param_list, 1,
+					baudrate);
+			if (ret < 0) {
+				LOG_ERR("AT parameter error");
+				return -EINVAL;
+			}
+		}
+		if (*baudrate == 1200) {
+			ret = 0;
+		} else if (*baudrate == 2400) {
+			ret = 0;
+		} else if (*baudrate == 4800) {
+			ret = 0;
+		} else if (*baudrate == 9600) {
+			ret = 0;
+		} else if (*baudrate == 14400) {
+			ret = 0;
+		} else if (*baudrate == 19200) {
+			ret = 0;
+		} else if (*baudrate == 38400) {
+			ret = 0;
+		} else if (*baudrate == 57600) {
+			ret = 0;
+		} else if (*baudrate == 115200) {
+			ret = 0;
+		} else if (*baudrate == 230400) {
+			ret = 0;
+		} else if (*baudrate == 460800) {
+			ret = 0;
+		} else if (*baudrate == 921600) {
+			ret = 0;
+		} else if (*baudrate == 1000000) {
+			ret = 0;
+		} else {
+			LOG_ERR("AT parameter error");
```

Comment:
The log message could be more specific here, `Invalid baud rate provided` or something like this.

Reply:
Modified to "Invalid uart baud rate provided."